src/engines/kotor/base/object.h

```diff
@@ -19,150 +19,126 @@
-
-	bool isClickable() const; ///< Can the player click the object?
-
-	// Positioning
-
 	/** Return the object's position within its area. */
 	virtual void getPosition(float &x, float &y, float &z) const;
 	/** Return the object's orientation. */
 	virtual void getOrientation(float &x, float &y, float &z, float &angle) const;
+	/** Get a room the object is in. */
+	const Room *getRoom();
+
+	/** Is the object's model(s) visible? */
+	virtual bool isVisible() const;
+	/** Is the object static (not manipulable at all)? */
+	bool isStatic() const;
+	/** Can the object be used by the PC? */
+	bool isUsable() const;
+	/** Can the player click the object? */
+	bool isClickable() const;
 
+	/** Set the maximum hit points for the objects. */
+	void setMaxHitPoints(int maxHP);
+	/** Set the current hitpoints. */
+	void setCurrentHitPoints(int hitpoints);
+	/** Set if the object has a minimum of one hp. */
+	void setMinOneHitPoints(bool enabled);
 	/** Set the object's position within its area. */
 	virtual void setPosition(float x, float y, float z);
 	/** Set the object's orientation. */
 	virtual void setOrientation(float x, float y, float z, float angle);
-
+	/** Set a room the object is in. */
+	void setRoom(const Room *room);
+
+	/** Show the object's model(s). */
+	virtual void show();
+	/** Hide the object's model(s). */
+	virtual void hide();
+	/** Hide the object's model(s) if applicable. */
+	virtual void hideSoft();
+	/** Make the object look at the specified location. */
 	void makeLookAt(float x, float y);
+	/** Make the object look at the specified object. */
 	void makeLookAt(Object *target);
-
-	/** Get a room the object is in. */
-	Room *getRoom();
-	/** Set a room the object is in. */
-	void setRoom(Room *room);
-
-	// Object/Cursor interactions
-
-	virtual void enter(); ///< The cursor entered the object.
-	virtual void leave(); ///< The cursor left the object.
-
-	/** (Un)Highlight the object. */
+	/** The cursor entered the object. */
+	virtual void enter();
+	/** The cursor left the object. */
+	virtual void leave();
+	/** (Un)highlight the object. */
 	virtual void highlight(bool enabled);
-
-	/** The object was clicked. */
+	/** The object is clicked on. */
 	virtual bool click(Object *triggerer = 0);
-
-	// Sound
-
 	/** Stop the current object sound. */
 	void stopSound();
 	/** Play an object sound. */
 	void playSound(const Common::UString &sound, bool pitchVariance = false);
 
-	// Animation
-
+	/** Play the object's model animation. */
 	virtual void playAnimation(const Common::UString &anim,
 	                           bool restart = true,
 	                           float length = 0.0f,
 	                           float speed = 1.0f);
 
 protected:
 	ObjectType _type; ///< The object's type.
-
-	Common::UString _name;        ///< The object's display name.
+	Common::UString _name; ///< The object's display name.
 	Common::UString _description; ///< The object's description.
-
```

Comment:
Can you keep the previous formatting style of the file, instead of removing all the grouping newlines?

Reply:
Okay. What about the grouping of methods though? I find it much easier to navigate class declarations when their mehods are grouped by type (getters, setters, actors, event handlers, etc.) rather then functionality.

Reply:
I find it more useful to group them according to functionality.